File: leetcode-04/solutions/cpp/0203.cpp
class Solution {
 public:
  ListNode* removeElements(ListNode* head, int val) {
    ListNode dummy(0);
    dummy.next = head;
    ListNode* curr = &dummy;

    while (curr) {
      ListNode* next = curr->next;
      while (next && next->val == val) next = next->next;
      curr->next = next;
      curr = next;
    }

    return dummy.next;
  }
};

File: leetcode-04/solutions/python3/0203.py
class Solution:
    def removeElements(self, head: ListNode, val: int) -> ListNode:
        dummy = ListNode(None)
        dummy.next = head
        curr = dummy

        while curr:
            next = curr.next
            while next and next.val == val:
                next = next.next
            curr.next = next
            curr = next

        return dummy.next
